frontend/src/components/InstallModal.js

```diff
@@ -29,10 +29,19 @@ class InstallModal extends React.Component {
   };
 
   componentDidUpdate(prevProps) {
-    const { operator } = this.props;
+    const { operator, history } = this.props;
+    const path = history.location.pathname.split('/');
+
+    let channelName;
+    const operatorId = _.get(operator, 'id');
+
+    if (path.length === 4) {
+      [channelName] = path.slice(2);
+    }
+
+    const installPath = `install/${channelName ? `${_.get(operator, 'channel')}/` : ''}${operatorId}.yaml`;
```

Comment:
The determination of the install path should only occur if the operator has changed.

Reply:
This is exact same code as before, just extracted to standalone method for code clarity. 
However yes, there could be 2 separate methods